common/src/main/java/com/wynntils/mc/event/DataComponentGetEvent.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © Wynntils 2025.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.mc.event;
+
+import net.minecraft.core.component.DataComponentType;
+import net.minecraft.core.component.DataComponents;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.bus.api.Event;
+
+public abstract class DataComponentGetEvent<T> extends Event {
+    private final ItemStack itemStack;
+    private final DataComponentType<T> dataComponentType;
+    private final T originalValue;
+    private T newValue;
+
+    protected DataComponentGetEvent(ItemStack itemStack, DataComponentType<T> dataComponentType, T originalValue) {
+        this.itemStack = itemStack;
+        this.dataComponentType = dataComponentType;
+        this.originalValue = originalValue;
+        this.newValue = originalValue;
+    }
+
+    public ItemStack getItemStack() {
+        return itemStack;
+    }
+
+    public DataComponentType<T> getDataComponentType() {
+        return dataComponentType;
+    }
+
+    public T getOriginalValue() {
+        return originalValue;
+    }
+
+    public T getNewValue() {
```

Comment:
```suggestion
    public T getValue() {
```

That's how we typically name things like this; the "new" is assumed from not being "Original".